dht_test.go

```diff
@@ -1075,3 +1076,39 @@ func TestFindClosestPeers(t *testing.T) {
 		t.Fatalf("got wrong number of peers (got %d, expected %d)", len(out), KValue)
 	}
 }
+
+func TestGetSetPluggedProtocol(t *testing.T) {
```

Comment:
I'd also test two nodes with *different* protocols.

Reply:
> I'd also test two nodes with different protocols.

What you like to see in such a test?

I could imagine creating two DHTs, A and B, each configured to use a unique protocol. We'd connect the two DHTs' hosts... and then verify that the A's routing table didn't contain B's peer ID (and vice-versa) after some fixed amount of time?

Reply:
Something like that. Or, even simpler, connect the DHTs then call Put on one and Get on the other.

Basically, I want to make sure they *can't* talk to eachother.

Reply:
Note: This isn't a critical test so don't waste too much time on it if it turns out to be tricky.

Reply:
> Or, even simpler, connect the DHTs then call Put on one and Get on the other.

done :)